ampd/src/tofnd/client.rs

```diff
@@ -2,156 +2,185 @@
 
 use async_trait::async_trait;
 use ecdsa::VerifyingKey;
-use error_stack::{FutureExt, IntoReport, Result, ResultExt};
-use mockall::automock;
-use tonic::transport::Channel;
-use tonic::{Response, Status};
+use error_stack::{FutureExt, IntoReport, ResultExt};
+use tokio::{sync::mpsc, sync::oneshot};
 
 use super::proto::{
-    key_presence_response::Response as KeyPresenceEnum, keygen_response::KeygenResponse,
-    multisig_client, sign_response::SignResponse, KeyPresenceRequest, KeygenRequest, SignRequest,
+    keygen_response::KeygenResponse, sign_response::SignResponse, KeygenRequest, SignRequest,
 };
-use super::{error::Error, error::TofndError, Signature};
+use super::{error::Error, error::TofndError, grpc::MultisigClient, Signature};
 use crate::types::PublicKey;
 
-#[automock]
+type Result<T> = error_stack::Result<T, Error>;
+
+type Handle<T> = oneshot::Sender<Result<T>>;
+
+pub enum Request {
+    Keygen {
+        request: KeygenRequest,
+        handle: Handle<PublicKey>,
+    },
+    Sign {
+        request: SignRequest,
+        handle: Handle<Signature>,
+    },
+}
+
 #[async_trait]
-pub trait MultisigClient {
-    async fn keygen(&mut self, request: KeygenRequest) -> Result<KeygenResponse, Status>;
-    async fn sign(&mut self, request: SignRequest) -> Result<SignResponse, Status>;
-    async fn key_presence(
-        &mut self,
-        request: KeyPresenceRequest,
-    ) -> Result<KeyPresenceEnum, Status>;
+pub trait Multisig {
```

Comment:
What's the purpose of having this trait?

Reply:
no need removed